mlir/lib/Dialect/Tosa/Transforms/TosaFoldCommon.cpp

```diff
@@ -34,16 +36,52 @@ DenseElementsAttr mlir::tosa::applyElementWise(
   // all of them to avoid dynamic resizing
   transformedValues.reserve(toTransform.getNumElements());
   for (auto val : toTransform.getValues<llvm::APFloat>()) {
-    auto recipVal = toApply(val, toTransform.getElementType());
-    transformedValues.push_back(recipVal);
+    auto transformedVal = toApply(val, toTransform.getElementType());
+    transformedValues.push_back(transformedVal);
   }
 
-  // Replace the current tensor with one containing the computed reciprocals
+  // Replace the current tensor with one containing the computed values
   auto newTensor =
       DenseElementsAttr::get(toTransform.getType(), transformedValues);
   return newTensor;
 }
 
+DenseElementsAttr mlir::tosa::applyElementWise(
+    const DenseElementsAttr &first, const DenseElementsAttr &second,
+    TensorType targetType,
+    const std::function<APFloat(const APFloat &, const APFloat &)> &toApply) {
+  // Make sure to use the correct values in case broadcasting is required
+  SmallVector<APFloat> transformedValues;
+  // We already know the amount of values we will insert, reserve space for
+  // all of them to avoid dynamic resizing
+  auto targetSize = 1;
+  auto targetShape = targetType.getShape();
+  for (const auto &dimSize : targetShape) {
+    targetSize *= dimSize;
+  }
+  transformedValues.reserve(targetSize);
+
```

Comment:
I wonder whether we should have a "fast path" that just does 
```
 for (auto offset = 0; offset < targetSize; offset++) {
    transformedValues.push_back(toApply(firstIt[offset ], secondIt[offset]););
  }
```
when no broadcasting is necessary.
But this is probably too premature optimization.

Reply:
I though about this as well. Maybe `getBroadcastedOffset` would be a nice location for this.